src/audio/module_adapter/module_adapter.c

```diff
@@ -1024,6 +1163,66 @@ static int module_adapter_audio_stream_type_copy(struct comp_dev *dev)
 	return ret;
 }
 
+#if CONFIG_ZEPHYR_DP_SCHEDULER
+static int module_adapter_copy_dp_queues(struct comp_dev *dev)
+{
+	/*
+	 * copy data from component audio streams to dp_queue
+	 * DP module processing itself will take place in DP thread
+	 * This is an adapter, to be removed when pipeline2.0 is ready
+	 */
+	struct processing_module *mod = comp_get_drvdata(dev);
+	int i = 0;
+	struct list_item *blist;
+
+	list_for_item(blist, &dev->bsource_list) {
+		/* input - we need to copy data from audio_stream (as source)
+		 * to dp_queue (as sink)
+		 */
+		struct comp_buffer *buffer =
+				container_of(blist, struct comp_buffer, sink_list);
+		struct comp_buffer __sparse_cache *buffer_c = buffer_acquire(buffer);
+		struct sof_source *data_src = audio_stream_get_source(&buffer_c->stream);
+
+		struct sof_sink *data_sink = dp_queue_get_sink(mod->dp_queue_ll_to_dp[i]);
+		uint32_t to_copy = MIN(sink_get_free_size(data_sink),
+				       source_get_data_available(data_src));
+
+		source_to_sink_copy(data_src, data_sink, true, to_copy);
+
+		buffer_release(buffer_c);
+		i++;
```

Comment:
remove 1194 and change 1187 to 
struct sof_sink *data_sink = dp_queue_get_sink(mod->dp_queue_ll_to_dp[i++]);
?
also for align, i = 0; put to 1177?